core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -320,7 +320,7 @@ private void commitCreateTransaction() {
 
     } catch (RuntimeException e) {
       // the commit failed and no files were committed. clean up each update
-      if (!ops.requireStrictCleanup() || e instanceof CleanableFailure) {
```

Comment:
Hm, isn't the existing logic correct? We should not clean up in strict mode unless it is `CleanableFailure`? Is there a chance the test in Trino produced a commit error that we treat as unknown?

Reply:
I think the one that failed in Trino was testing `SchemaNotFoundException` [here](https://github.com/trinodb/trino/blob/04cd049fbbc7f43ee827974650ae90d0b3357d19/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileMetastoreCreateTableFailure.java#L102).